src/View/Helper/FormHelper.php

```diff
@@ -2123,6 +2134,7 @@ public function multiCheckbox(string $fieldName, iterable $options, array $attri
                 'value' => '',
                 'secure' => false,
                 'disabled' => $attributes['disabled'] === true || $attributes['disabled'] === 'disabled',
+                'id' => $attributes['id'] . '-',
```

Comment:
Won't this append `-` to `true`?

Reply:
`_initInputField` will generate an ID if it's set to `true`.

But why append a dash in the first place? If there's no actual technical need for it, then we shouldn't impose such an atypical convention for people to remember, not least because it kinda looks like something hasn't been generated correctly.

Reply:
Also, this is not exactly BC, since people could be relying on the generated ID for JS/CSS and other operations
If that ID is not changed in a bugfix release, functionality might break.

Reply:
The goal of this change is to re-instate the id attribute on the hidden input so that FormHelper generates markup more consistent with 2.x. More important than the compatibility with 2.x, having an id on the hidden input enables easy to build ajax validation for radio buttons.

Reply:
> Won't this append `-` to `true`?

Even I was thinking need of this `-`, I added this as per CakePHP 2.x 

![2022-04-28_082854](https://user-images.githubusercontent.com/325154/165667822-ad640766-c675-4486-8cd6-358cba07f0db.jpg)

I can test Form helper test cases by removing it?

Reply:
Don't we run a higher risk of conflicting with other `id` attributes on pages? The value `user-id-` is much less likely to be used by application developers vs `user-id` which is a reasonable value for developers to use.

Reply:
Doesn't the same risk exist for any other field? That's why we have the id prefixing feature.

Reply:
That's what I'd say too. Sure, there's a slight chance for a clash with existing code, as previously no ID matching the field has been generated, but in the grand scheme of things it's no different than the ID for any other field.

Reply:
Fair enough. In a minor release we can document the new id attributes.

Reply:
I have removed trailing `-` from id and tested test cases successfully.